Framework/Algorithms/test/MergeRunsTest.h

```diff
@@ -1732,4 +1732,37 @@ class MergeRunsTest : public CxxTest::TestSuite {
   }
 };
 
+class MergeRunsTestPerformance : public CxxTest::TestSuite {
+public:
+  static MergeRunsTestPerformance *createSuite() {
+    return new MergeRunsTestPerformance();
+  }
+  static void destroySuite(MergeRunsTestPerformance *suite) { delete suite; }
+
+  MergeRunsTestPerformance() {}
+
+  void test_merge_detector_scan_workspaces() {
+    for (size_t i = 0; i < 10; ++i) {
+      // Create a D2B type workspace
+      const auto &ws = WorkspaceCreationHelper::
+          create2DDetectorScanWorkspaceWithFullInstrument(10000, 1, 25,
+                                                          i * 1000);
+      std::string wsName = "a" + std::to_string(i);
+      AnalysisDataService::Instance().addOrReplace(wsName, ws);
+    }
+
+    Mantid::Algorithms::MergeRuns mergeRuns;
+    mergeRuns.initialize();
+    mergeRuns.setPropertyValue("InputWorkspaces",
+                               "a0, a1, a2, a3, a4, a5, a6, a7, a8, a9");
+    mergeRuns.setPropertyValue("OutputWorkspace", "outputWS");
+    mergeRuns.execute();
```

Comment:
Just a comment, shouldn't only `execute()` be in the test body, the rest in setup/teardown?

Reply:
Yeah, it probably should, it is a smaller amount of time to do the setup but not insignificant.